### DISCUSSION TITLE
Add Support for Piwik and/or Google Tag Manager

### DISCUSSION THREAD

**Discussion by jason-warpstream** [GitbookIO/community#696] [Feature Requests]
**What I'm Requesting**

Given users cannot add custom JavaScript to global `head` or `body` tags of Gitbook and must use official integrations, I'm requesting integrations be investigated (and hopefully) made for [Piwik Pro](https://piwik.pro/) and [Google Tag Manager](https://tagmanager.google.com/).

**Desired Outcome**

Ideally, there would be official integrations for both these products where you just have to input your account ID and they work.

**The Impact and Why I'm Requesting These**

For Piwik:

- Many companies use it as an analytics alternative to Google Analytics.
- Without Piwik integration in Gitbook, gap exists in scenarios where someone leaves the main website to go to a docs website. Once on the docs website, no data can be sent to Piwik to see how that visitor interacted with Gitbook-hosted docs.

For Google Tag Manager:

- This is probably the most-used tag management platform, allowing code snippets to be hosted in a container so that code does not need to be hard-coded into webpages or a product and wait for a release.
- Adding support for Google Tag Manager would allow users to integrate other products easily as many SaaS products simply rely on firing a JavaScript snippet and linking it to an account ID.

**Comment by addisonschultz** [comment]
Hi @jason-warpstream - Great suggestions - I've moved these requests to our integration platform, where we track requests like these.

You can follow up or add more context here: https://github.com/GitbookIO/integrations/issues/532

**Reply by jason-warpstream** [reply]
Thanks, @addisonschultz. I'll follow over there.
